### ISSUE TITLE
Discrete scale doesn't work for datetime data.

### ISSUE THREAD

**Issue by AndreiKingsley** [JetBrains/lets-plot-kotlin#231]
With default (continuous) scale it works well:
```
import java.time.Instant.parse

val instants = listOf(
    "2021-01-01T00:00:00Z",
    "2021-02-23T00:00:00Z",
    "2021-04-03T00:00:00Z",
    "2021-05-04T00:00:00Z",
    "2021-10-05T00:00:00Z"
).map(::parse)

val daysData = mapOf<String, Any>(
    "days" to instants,
    "val" to (1..instants.size)
)

letsPlot(daysData) + geomBar() { x = "days"; fill = "days" }
```
<img width="605" alt="image" src="https://github.com/JetBrains/lets-plot-kotlin/assets/49092338/57f9280c-9eba-4fd7-bd64-fa51ce53339e">

But with discrete (via `asDiscrete()` or `scaleFillDiscrete()`) formatting is missing (in legend and tooltips)
```
letsPlot(daysData) + geomBar() { x = "days"; fill = asDiscrete("days") }
```
<img width="605" alt="image" src="https://github.com/JetBrains/lets-plot-kotlin/assets/49092338/226ce1ce-fdf5-47f8-afab-27ed17a583a8">

```
letsPlot(daysData) + geomBar() { x = "days"; fill = "days" } + scaleFillDiscrete()
```
<img width="609" alt="image" src="https://github.com/JetBrains/lets-plot-kotlin/assets/49092338/3d56aa09-108d-4d39-90a8-5ad3b0f4f9b2">

**Comment by OLarionova-HORIS** [comment]
Fixed via commit: https://github.com/JetBrains/lets-plot/commit/afefca4f0bf3b3f9f8a1cd6d09af804ea76493ea

**Comment by alshan** [comment]
Fixed in v4.7.1
